test/foundry/Confiscate.t.sol

```diff
@@ -0,0 +1,290 @@
+        fuzzedKeyAlice = bound(fuzzedKeyAlice, 1, SECP256K1_ORDER - 1);
+        address alice = vm.addr(fuzzedKeyAlice);
+        // Ensure the fuzzed key is within the valid range for secp256k1
+        fuzzedKeyBob = bound(fuzzedKeyBob, 1, SECP256K1_ORDER - 1);
+        address bob = vm.addr(fuzzedKeyBob);
+
+        vm.assume(alice != bob);
+        // Prank as Alice for the transaction
+        vm.startPrank(alice);
+        OffchainAssetReceiptVault vault = OffchainAssetVaultCreator.createVault(factory, alice, assetName, assetSymbol);
+
+        vault.grantRole(vault.CONFISCATOR(), alice);
+
+        // Stop recording logs
+        Vm.Log[] memory logs = vm.getRecordedLogs();
+        vault.confiscateShares(bob, justification);
+
+        // Check the logs to ensure event is not present
+        bool eventFound = false;
+
+        for (uint256 i = 0; i < logs.length; i++) {
+            if (logs[i].topics[0] == ConfiscateShares.selector) {
+                eventFound = true;
+                break;
+            }
+        }
+
+        assertFalse(eventFound, "ConfiscateShares event should not be emitted");
+        vm.stopPrank();
+    }
+
+    /// Test to checks ConfiscateShares
+    function testConfiscateShares(
+        uint256 fuzzedKeyAlice,
+        uint256 fuzzedKeyBob,
+        uint256 shareRatio,
+        uint256 aliceAssets,
+        string memory assetName,
+        string memory assetSymbol,
+        bytes memory justification,
+        uint256 certifyUntil,
+        uint256 referenceBlockNumber
+    ) external {
+        shareRatio = bound(shareRatio, 1, 1e18);
+        // Ensure the fuzzed key is within the valid range for secp256k1
+        fuzzedKeyAlice = bound(fuzzedKeyAlice, 1, SECP256K1_ORDER - 1);
+        address alice = vm.addr(fuzzedKeyAlice);
+        // Ensure the fuzzed key is within the valid range for secp256k1
+        fuzzedKeyBob = bound(fuzzedKeyBob, 1, SECP256K1_ORDER - 1);
+        address bob = vm.addr(fuzzedKeyBob);
+
+        referenceBlockNumber = bound(referenceBlockNumber, 1, block.number);
+        certifyUntil = bound(certifyUntil, 1, 1e6 - 1);
+
+        vm.assume(alice != bob);
+        // Prank as Alice for the transaction
+        vm.startPrank(alice);
+        OffchainAssetReceiptVault vault = OffchainAssetVaultCreator.createVault(factory, alice, assetName, assetSymbol);
+        vault.grantRole(vault.CONFISCATOR(), alice);
+        vault.grantRole(vault.DEPOSITOR(), alice);
+        vault.grantRole(vault.CERTIFIER(), alice);
+
+        // Call the certify function
+        vault.certify(certifyUntil, referenceBlockNumber, false, justification);
+
+        //set upperBound for assets so it does not overflow while calculating fixedPointDiv or fixedPointMul
+        uint256 upperBound = type(uint256).max / 1e18;
+        // Assume that aliceAssets is less than totalSupply
+        aliceAssets = bound(aliceAssets, 1, upperBound);
```

Comment:
why not 0?

Reply:
I'm depositing alliceAssets. if its 0 will get ZeroAssetsAmount() error. this case is tested in deposits tests